src/lib/AsCallback.coffee

```diff
@@ -0,0 +1,180 @@
+  received = []
+  outPorts = Object.keys network.graph.outports
+  outSockets = {}
+  outPorts.forEach (outport) ->
+    outSockets[outport] = internalSocket.createSocket()
+    process.component.outPorts[outport].attach outSockets[outport]
+    outSockets[outport].on 'ip', (ip) ->
+      res = {}
+      res[outport] = ip
+      received.push res
+  # Subscribe network finish
+  network.once 'end', ->
+    # Clear listeners
+    for port, socket of outSockets
+      process.component.outPorts[port].detach socket
+    outSockets = {}
+    inSockets = {}
+    callback null, received
+  # Start network
+  network.start (err) ->
+    return callback err if err
+    # Send inputs
+    for inputMap in inputs
+      for port, value of inputMap
+        if IP.isIP value
+          inSockets[port].post value
+          continue
+        inSockets[port].post new IP 'data', value
+
+getType = (inputs, network) ->
+  # Scalar values are always simple inputs
+  return 'simple' unless typeof inputs is 'object'
+
+  if Array.isArray inputs
+    maps = inputs.filter (entry) ->
+      getType(entry, network) is 'map'
+    # If each member if the array is an input map, this is a sequence
+    return 'sequence' if maps.length is inputs.length
+    # Otherwise arrays must be simple inputs
+    return 'simple'
+
+  # Empty objects can't be maps
+  return 'simple' unless Object.keys(inputs).length
+  for key, value of inputs
+    return 'simple' unless network.graph.inports[key]
+  return 'map'
+
+prepareInputMap = (inputs, inputType, network) ->
+  # Sequence we can use as-is
+  return inputs if inputType is 'sequence'
+  # We can turn a map to a sequence by wrapping it in an array
+  return [inputs] if inputType is 'map'
+  # Simple inputs need to be converted to a sequence
+  inPort = Object.keys(network.graph.inports)[0]
+  # If we have a port named "IN", send to that
+  inPort = 'in' if network.graph.inports.in
+  map = {}
+  map[inPort] = inputs
+  return [map]
+
+normalizeOutput = (values, options) ->
+  return values if options.raw
+  result = []
+  previous = null
+  current = result
+  for packet in values
+    if packet.type is 'openBracket'
+      previous = current
+      current = []
+      previous.push current
+    if packet.type is 'data'
+      current.push packet.data
+    if packet.type is 'closeBracket'
+      current = previous
+  if result.length is 1
+    return result[0]
+  return result
+
+sendOutputMap = (outputs, resultType, options, callback) ->
+  # First check if the output sequence contains errors
+  errors = outputs.filter((map) -> map.error?).map (map) -> map.error
+  return callback normalizeOutput errors, options if errors.length
+
+  if resultType is 'sequence'
+    return callback null, outputs.map (map) ->
+      res = {}
+      for key, val of map
+        if options.raw
+          res[key] = val
+          continue
+        res[key] = normalizeOutput [val], options
+      return res
+
+  # Flatten the sequence
```

Comment:
This behavior should be documented and/or there should be an option to return unflattened output. For those who care which ports are yielded together.